compiler/src/dotty/tools/dotc/transform/localopt/ConstantFold.scala

```diff
@@ -181,22 +181,19 @@ import ast.Trees._
         case _ => false
       }
     case t1: Ident =>
-      desugarIdent(t1) match {
-        case Some(t) =>
-          val t2i = t2 match {
-            case t2: Ident => desugarIdent(t2).getOrElse(t2)
-            case _ => t2
-          }
-          isSimilar(t, t2i)
-        case None => t1.symbol eq t2.symbol
```

Comment:
I think the logic was wrong here. If `t2` is a select then it is not enough to compare the symbol